src/videojs-http-streaming.js

```diff
@@ -737,17 +737,21 @@ class HlsHandler extends Component {
   setupQualityLevels_() {
     let player = videojs.players[this.tech_.options_.playerId];
 
-    if (player && player.qualityLevels) {
-      this.qualityLevels_ = player.qualityLevels();
+    // if there isn't a player or there isn't a qualityLevels plugin
+    // or qualityLevels_ listeners have already been setup, do nothing.
+    if (!player || !player.qualityLevels || this.qualityLevels_) {
```

Comment:
Only setup quality levels listeners once, the rest of the code is the same except we return early instead of adding the logic in an `if` scope.